lib/api/api.js

```diff
@@ -184,14 +186,103 @@ const api = {
             }
             return { returnTagCount, isImplicitDeny };
         }
+        let formDataParser;
+        let fileEventData = null;
+
+        if (apiMethod === 'objectPost') {
+            formDataParser = busboy({ headers: request.headers });
+        }
 
         return async.waterfall([
+            next => {
+                if (apiMethod === 'objectPut' || apiMethod === 'objectPutPart') {
+                    return next(null);
+                }
+                if (apiMethod === 'objectPost' && request.headers['content-type'].includes('multipart/form-data')) {
+                    writeContinue(request, response);
+
+                    request.formData = {};
+                    let totalFieldSize = 0;
+                    const MAX_FIELD_SIZE = 20 * 1024; // 20KB
+                    formDataParser.on('field', (fieldname, val) => {
+                        totalFieldSize += Buffer.byteLength(val, 'utf8');
+                        if (totalFieldSize > MAX_FIELD_SIZE) {
+                            const err = errors.MaxPostPreDataLengthExceeded;
+                            formDataParser.emit('error', err);
+                        }
+                        // Convert fieldname to lowercase for case-insensitive comparison
+                        const lowerFieldname = fieldname.toLowerCase();
+                        request.formData[lowerFieldname] = val;
+                    });
+
+                    formDataParser.on('file', (fieldname, file, filename, encoding, mimetype) => {
+                        fileEventData = { fieldname, file, filename, encoding, mimetype };
+                        return next(null);
+                    });
+
+                    formDataParser.on('finish', () => {
+                        // No file field == error
+                        if (!fileEventData) {
+                            const err = errors.InvalidArgument
+                                .customizeDescription('POST requires exactly one file upload per request.');
+                            formDataParser.emit('error', err);
+                        }
+                        return undefined;
+                    });
+
+                    formDataParser.on('error', (err) => {
+                        log.trace('Error processing form data:', {
+                            error: err,
```

Comment:
this right now produce this message: `"Error processing form data:"`
You should stringify or show the `message` of the error

Reply:
realize I didn't properly address this, adding here:
99b0c26d626a84ba03cd28969660708d3d15e3a0

Reply:
Note that `.toString()` can be interesting as it also has the name of the error before the message

Reply:
improved with toString
7c183269cdcbb468292f759307a28b4e7c58a210